oximeter/oximeter/schema/instance-data-link.toml

```diff
@@ -0,0 +1,86 @@
+format_version = 1
+
+[target]
+name = "instance_data_link"
+description = "A network data link attached to a virtual machine instance"
+authz_scope = "fleet"
+versions = [
+    { version = 1, fields = [ "interface_id", "instance_id", "project_id", "silo_id", "sled_id", "sled_model", "sled_revision", "sled_serial" ] },
+]
+
+[[metrics]]
+name = "bytes_sent"
+description = "Total number of bytes sent on the data link"
+units = "bytes"
+datum_type = "cumulative_u64"
+versions = [
+    { added_in = 1, fields = [] }
+]
+
+[[metrics]]
+name = "bytes_received"
+description = "Total number of bytes received on the data link"
+units = "bytes"
+datum_type = "cumulative_u64"
+versions = [
+    { added_in = 1, fields = [] }
+]
+
+[[metrics]]
+name = "packets_sent"
+description = "Total number of packets sent on the data link"
+units = "count"
+datum_type = "cumulative_u64"
+versions = [
+    { added_in = 1, fields = [] }
+]
+
+[[metrics]]
+name = "packets_received"
+description = "Total number of packets received on the data link"
+units = "count"
+datum_type = "cumulative_u64"
+versions = [
+    { added_in = 1, fields = [] }
+]
+
+[[metrics]]
+name = "packets_dropped"
+description = "Number of packets dropped on the RX queue of the data link"
+units = "count"
+datum_type = "cumulative_u64"
+versions = [
+    { added_in = 1, fields = [] }
+]
+
+[fields.interface_id]
+type = "uuid"
+description = "The ID of the data link"
```

Comment:
Sorry for all the back and forth on this one, I appreciate your patience.

Seeing this called `interface_id` with a description referring to a "data link" makes me wonder if we've gotten the target name wrong. We picked `instance_data_link` to match up with the `sled_data_link` and `switch_data_link` targets. That's nice, except the public API doesn't refer to "data links" anywhere. As this field name indicates, the customer operates on `NetworkInterface` objects through the API. I worry it'd be a bit confusing to say, "Hey, I wonder what this interface is doing? Let me go look for a timeseries about 'interfaces'", only to find out that the timeseries is really `instance_data_link:bytes_sent` or whatever. I think I'd find that annoying or confusing, learning it for the first time.

What do you think about renaming the target to `instance_network_interface` or `virtual_network_interface`, or possibly just `network_interface`?

Reply:
Yeah, keeping to the naming around instance-data-link has been confusing. I'd say just `network_interface` and therefore `network-interface.toml`, right? 

Then this description is "The ID of the network interface". Thoughts @bnaecker?

Reply:
Sweet, yeah the description sounds right in that case. I would personally vote for `instance_network_interface`, just to avoid any possibility of confusion. That sound OK to you?

Reply:
Oh, and I would `s/data link/network interface` in the other descriptions too.

Reply:
Sure, `instance_network_interface` it is. I'll push up a new PR.